### PR TITLE
Fix Order#empty! update order's state to cart if order not completed …

### PR DESCRIPTION
**Pull request by tanmay3011**
…else raise error

### CODE REVIEW

File: core/app/models/spree/order.rb [hunk 1]
```diff
@@ -438,14 +438,20 @@ def ensure_line_items_are_in_stock
     end
 
     def empty!
-      line_items.destroy_all
-      updater.update_item_count
-      adjustments.destroy_all
-      shipments.destroy_all
-      state_changes.destroy_all
+      if completed?
+        raise 'Order has already been processed so it cannot be emptied'
```
Comment (priyank-gupta) [inline core/app/models/spree/order.rb]: @tanmay3011 I think we should move this to I18n.

**Comment by tanmay3011** [comment]
@priyank-gupta  Updated, Please review.

**Comment by priyank-gupta** [comment]
:+1: 
Create a direct PR and then close this.

**Comment by tanmay3011** [comment]
Merged

**Pull request closed without merging**